File: problems/01_multiples_of_3_or_5/multiples_of_3_or_5.py
# multiples_of_3_or_5.py
# Straightforward scan; final print redacted.


def is_multiple_of_3_or_5(n: int) -> bool:
    return n % 3 == 0 or n % 5 == 0


def sum_scan(limit: int) -> int:
    total = 0
    for k in range(1, limit):  # exclusive upper bound
        if is_multiple_of_3_or_5(k):
            total += k
    return total


def main() -> None:
    limit = 1000
    result = sum_scan(limit)
    # print(result)  # TODO: reveal locally
    print(f"Sum for {limit} = <compute locally>")
    # Example from the statement:
    assert sum_scan(10) == 23
    _ = result


if __name__ == "__main__":
    main()

File: problems/01_multiples_of_3_or_5/multiples_of_3_or_5.cpp
// multiples_of_3_or_5.cpp
// Simple readable implementation; final print redacted.

#include <cstdint>
#include <iostream>

static bool is_multiple_of_3_or_5(std::uint64_t n) {
    return (n % 3 == 0) || (n % 5 == 0);
}

static std::uint64_t sum_scan(std::uint64_t limit) {
    std::uint64_t total = 0;
    for (std::uint64_t k = 1; k < limit; ++k) {
        if (is_multiple_of_3_or_5(k)) {
            total += k;
        }
    }
    return total;
}

int main() {
    const std::uint64_t limit = 1000;
    const auto result = sum_scan(limit);
    // std::cout << "Sum for " << limit << " = " << result << "\n"; // TODO: reveal locally
    std::cout << "Sum for " << limit << " = <compute locally>\n";
    // Example check: below 10 -> 23
    if (sum_scan(10) != 23) {
        return 1;
    }
    (void)result;
    return 0;
}
